From 0483f14697ac8da2543a0faa9283417b892105fe Mon Sep 17 00:00:00 2001
From: Carlos Munoz <carlos.munoz@caviumnetworks.com>
Date: Thu, 16 Jan 2014 17:09:20 -0800
Subject: [PATCH 371/518] spi: spi-octeon.c: Allow faster speeds on octeonIII
 boards.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 769e36ac53797c9710d4291861c3ae47fb00ba31
Description:

Signed-off-by: Carlos Munoz <carlos.munoz@caviumnetworks.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/spi/spi-octeon.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/spi-octeon.c b/drivers/spi/spi-octeon.c
index 24daf96..b881dfb 100644
--- a/drivers/spi/spi-octeon.c
+++ b/drivers/spi/spi-octeon.c
@@ -25,7 +25,8 @@
 
 #define OCTEON_SPI_MAX_BYTES 9
 
-#define OCTEON_SPI_MAX_CLOCK_HZ 16000000
+#define OCTEON_SPI_MAX_CLOCK_HZ		16000000
+#define OCTEON_SPI_MAX_CLOCK_HZ_7XXX	200000000	/** Max clock for O3 */
 
 struct octeon_spi {
 	struct spi_master *my_master;
@@ -83,8 +84,14 @@ static int octeon_spi_do_transfer(struct octeon_spi *p,
 	if (xfer->bits_per_word)
 		bits_per_word = xfer->bits_per_word;
 
-	if (speed_hz > OCTEON_SPI_MAX_CLOCK_HZ)
-		speed_hz = OCTEON_SPI_MAX_CLOCK_HZ;
+	if (OCTEON_IS_OCTEON3()) {
+		if (speed_hz > OCTEON_SPI_MAX_CLOCK_HZ_7XXX)
+			speed_hz = OCTEON_SPI_MAX_CLOCK_HZ_7XXX;
+	}
+	else {
+		if (speed_hz > OCTEON_SPI_MAX_CLOCK_HZ)
+			speed_hz = OCTEON_SPI_MAX_CLOCK_HZ;
+	}
 
 	clkdiv = octeon_get_io_clock_rate() / (2 * speed_hz);
 
-- 
1.9.1

